source/common/quic/envoy_quic_client_stream.cc

```diff
@@ -110,8 +110,6 @@ void EnvoyQuicClientStream::resetStream(Http::StreamResetReason reason) {
 }
 
 void EnvoyQuicClientStream::switchStreamBlockState(bool should_block) {
```

Comment:
This assert was basically "we should not be flow control blocked before reading body data" which turns out not to be true once you have multiple streams on one connection.